modules/library/main/src/main/java/org/geotools/filter/function/InterpolateFunction.java

```diff
@@ -301,28 +301,22 @@ public Object evaluate(Object object) {
         return evaluate(object, Object.class);
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * <p>When {@code context} is unspecified (i.e. {@code null} or {@code Object.class}), it'll be
+     * derived from the {@code method} parameter, as {@code java.awt.Color.class} when {@code method
+     * == COLOR}, and as {@code java.lang.Double} when {@code method == NUMERIC}.
+     *
+     * @throws IllegalArgumentException if the {@code method} parameter is {@code NUMERIC} and
+     *     {@code context == java.awt.Color.class}, or {@code method} is {@code COLOR} and {@code
+     *     context} is specified and is not {@code context == java.awt.Color.class}
```

Comment:
Hold up, we should be able to get a String out of a colour. We went to a lot of trouble with converters to do so ...
We went to a lot of trouble to with converters to avoid being strongly typed ...

Reply:
Yeah you're right... the original implementation was rigid, but functions should try to use converters to go from the internal representation they come up with, to the desired output, and return null if they fail (see Expression javadoc). 
Converters are also pluggable, so it's not possible to know in advance what would work, and what wouldn't, one just has to try.

Reply:
Not sure I understand. What changed is it'll output Color or Double if no context was provided. It changes nothing otherwise and the interpolate functions still return `Converters.convert(interpolated, context)`, where `context` hasn't been changed if it was `String.class` or anything else but `null` or `Object.java`. both cases were failing.

Reply:
What Jody is saying is that the original implementation is broken in more ways than you've fixed.